docs: fix return type's array shape for `WP_Speculation_Rules::jsonSerialize()`

justlevine opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

This PR fixes the return type on `WP_Speculation_Rules::jsonSerialize()` to show that it returns an `array<string, mixed[]>` instead of the internal array having a `string` key. This is due to the method's use of `array_values()` to intentionally strip any keys.

While this issue was surfaced via PHPStan in https://github.com/WordPress/wordpress-develop/pull/7619 (trac: https://core.trac.wordpress.org/ticket/61175 ), it can be remediated independently of that ticket.

Trac ticket:
https://core.trac.wordpress.org/ticket/64238
https://core.trac.wordpress.org/ticket/63268

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

justlevine committed:
docs: fix return type for WP_Speculation_Rules::jsonSerialize to array<string,list>

justlevine commented:
Not sure if this warrants a 6.8.x commit, but I definitely don't think it's worth bothering to create a Trac ticket just to make that happen.

johnbillion commented on src/wp-includes/class-wp-speculation-rules.php:
```diff
@@ -230,7 +230,7 @@ public function has_rule( string $mode, string $id ): bool {
 	 * @since 6.8.0
 	 *
-	 * @return array<string, array<string, mixed>> Speculation rules data.
+	 * @return array<string, mixed[]> Speculation rules data.
```
How about `array<string, array<int, mixed>>`?

justlevine replied:
```suggestion
	 * @return array<string, array<int ,mixed>> Speculation rules data.
```

justlevine replied:
(GH autoresolved that on commit - repoened)

justlevine replied:
Resolved in a322456d2eb0784f5e4f0dddb195b829f9895ab4

westonruter replied:
The `mixed` here is a rule which is actually an `array<string, mixed>`. I've applied that in f5700b99067f3dee4e004bbe9b749ee8a56b6993.

justlevine committed:
Update src/wp-includes/class-wp-speculation-rules.php

Per https://github.com/WordPress/wordpress-develop/pull/8946#discussion_r2143826139

westonruter committed:
Merge branch 'trunk' into docs/WP_Speculation_Rules--jsonSerialize-return-type

westonruter committed:
Use array<string, mixed> to type a rule instead of just mixed

westonruter commented:
Committed in r61280 (9e2d045)
